src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -34,86 +40,100 @@ public CertificateCallbackMapper(Func<HttpRequestMessage, X509Certificate2, X509
 
         public static async ValueTask<(Socket, Stream)> ConnectAsync(string host, int port, CancellationToken cancellationToken)
         {
-            try
+            // Rather than creating a new Socket and calling ConnectAsync on it, we use the static
+            // Socket.ConnectAsync with a SocketAsyncEventArgs, as we can then use Socket.CancelConnectAsync
+            // to cancel it if needed. Rent or allocate one.
+            ConnectEventArgs saea;
+            if (!s_connectEventArgs.TryDequeue(out saea))
             {
-                // Rather than creating a new Socket and calling ConnectAsync on it, we use the static
-                // Socket.ConnectAsync with a SocketAsyncEventArgs, as we can then use Socket.CancelConnectAsync
-                // to cancel it if needed.
-                using (var saea = new BuilderAndCancellationTokenSocketAsyncEventArgs(cancellationToken))
-                {
-                    // Configure which server to which to connect.
-                    saea.RemoteEndPoint = IPAddress.TryParse(host, out IPAddress address) ?
-                        (EndPoint)new IPEndPoint(address, port) :
-                        new DnsEndPoint(host, port);
+                saea = new ConnectEventArgs();
+            }
+            saea.Initialize(cancellationToken);
```

Comment:
Q about style: I thought that we usually added a blank line after a block, no?

Reply:
Yeah, we generally do, and I probably should have here.